src/main/arm/createUiDefinition.json

```diff
@@ -10,15 +10,15 @@
                 "visible": "[not(equals(length(basics('identity').userAssignedIdentities), 1))]",
                 "options": {
                     "icon": "Error",
-                    "text": "Select <b>only one</b> user-assigned managed identity that has a <b>Contributor</b> role in the subscription.<br><br>You can find more details from the following articles:<li><a href='https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-manage-user-assigned-managed-identities?pivots=identity-mi-methods-azp#create-a-user-assigned-managed-identity' target='_blank'>Create a user-assigned managed identity</a></li><li><a href='https://docs.microsoft.com/azure/role-based-access-control/role-assignments-portal-managed-identity#user-assigned-managed-identity' target='_blank'>Assign a role to a user-assigned managed identity</a></li>"
+                    "text": "Select <b>only one</b> user-assigned managed identity that has an <b>Owner</b> role or <b>Contributor</b> and <b>User Access Administrator</b> roles in the subscription and a <b>Directory readers</b> role in Azure AD.<br><br>You can find more details from the following articles:<li><a href='https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-manage-user-assigned-managed-identities?pivots=identity-mi-methods-azp#create-a-user-assigned-managed-identity' target='_blank'>Create a user-assigned managed identity</a></li><li><a href='https://docs.microsoft.com/azure/role-based-access-control/role-assignments-portal-managed-identity#user-assigned-managed-identity' target='_blank'>Assign a role to a user-assigned managed identity</a></li><li><a href='https://docs.microsoft.com/azure/active-directory/roles/manage-roles-portal#assign-a-role' target='_blank'>Assign Azure AD roles to a user-assigned managed identity</a></li>"
```

Comment:
Hello @git4rk what do you think about prepending the following sentence to the text on line 13?

```html
<b>The deployment will fail unless the roles specified here are assigned to the user-assigned managed identity.</b>
```

Reply:
@edburns yes, good idea.
Looking at the text again, I think we should also clarify the roles:

Select <b>only one</b> user-assigned managed identity that has `(1)` an <b>Owner</b> role or <b>Contributor</b> and <b>User Access Administrator</b> roles in the subscription, and `(2)` a <b>Directory readers</b> role in Azure AD.<br>

Reply:
Thanks for your feedback. The comments look good to me, and they're addressed in commit 6a54a20.
Here is the screenshot where the changes are highlighted:
![image](https://user-images.githubusercontent.com/10357495/151688376-b7706f8c-35dd-48af-8fc9-132a0e1b48b5.png)